### PR TITLE
tasks

### PR DESCRIPTION
**Pull request by meghnaneelayavalasa**
2.1-2.5

### CODE REVIEW

File: 2.1/Basic Mathematical Operations.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+        case '+':
+            return value1 + value2;
+        case '-':
+            return value1 - value2;
+        case '*':
+            return value1 * value2;
+        case '/':
+            return value1 / value2;
+        default:
+            return 0;
```
Comment (SurajPareek01) [inline 2.1/Basic Mathematical Operations.js:14]: Can we have the devide by zero check here?

File: 2.1/Transportation on vacation.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+//Transportation on vacation
+function rentalCarCost(d) {
+  // Your solution here
+  if(d>=7)
```
Comment (SurajPareek01) [inline 2.1/Transportation on vacation.js]: Please avoid using of magic numbers;

File: 2.2/Middle Character.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+//Get the Middle Character
+function getMiddle(s)
+{
+  var position;
+  var length;
+  
+  if(s.length % 2 == 1) {
```
Comment (SurajPareek01) [inline 2.2/Middle Character.js]: please use strict type checking ( === )

File: 2.3/Can you keep a secret.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+//Can you keep a secret?
+function createSecretHolder(secret) {
+  return {
+ getSecret : function() {
```
Comment (SurajPareek01) [inline 2.3/Can you keep a secret.js]: please beautify this code.

File: 2.4/A Chain adding function.js [hunk 5]
```diff
@@ -0,0 +1,6 @@
+// Let the currying begin!
+  var add = function(n) {
+  const f = x => add(n + x)
```
Comment (SurajPareek01) [inline 2.4/A Chain adding function.js]: please beautify this code

File: 2.3/prefill an array.js [hunk 6]
```diff
@@ -0,0 +1,8 @@
+function prefill(n, v) {
+    if(typeof n === 'boolean' || !isFinite(n) || !Number.isInteger(Number(n)) || Number(n) < 0)
+        throw new TypeError(n + " is invalid")
+    arr = []
```
Comment (SurajPareek01) [inline 2.3/prefill an array.js]: please use semi-colon;

File: 2.2/Middle Character.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+      position = s.length / 2;
+      length = 1;
+  } 
+  else {
+      position = s.length / 2 - 1;
+      length = 2;
+  }
+  return s.substring(position, position + length)
```
Comment (SurajPareek01) [inline 2.2/Middle Character.js]: please use semi-colon;

File: 2.4/Using closures to share class state.js [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+  const constructor = function (name, weight) {
+    if (!name || !weight) throw Error('invalid parameters')
+    Object.defineProperty(this, 'name', { get: () => name })
+    Object.defineProperty(this, 'weight', { get: () => weight, set: value => weight = value })
+    cats.push(this)
+  }
+  constructor.averageWeight = function() {
+      sum = 0
+      for(i in cats){
+        sum += cats[i].weight
```
Comment (SurajPareek01) [inline 2.4/Using closures to share class state.js]: please use semi-colon

File: 2.5/Array_Helpers.js [hunk 9]
```diff
@@ -0,0 +1,14 @@
+Array.prototype.square = function () { return this.map(function (n) { return n * n; }); }
```
Comment (SurajPareek01) [inline 2.5/Array_Helpers.js:1]: Where is 'numbers' defined? It won't work.

**Review by SurajPareek01 — CHANGES_REQUESTED** [review]

**Review by SurajPareek01 — APPROVED** [review]

**Comment by meghnaneelayavalasa** [comment]
Reviewed

**Pull request closed without merging**